src/snap-confine.c

```diff
@@ -113,6 +113,11 @@ int main(int argc, char **argv)
 			setup_devices_cgroup(security_tag, &udev_s);
 		snappy_udev_cleanup(&udev_s);
 
+		// Setup the XDG_RUNTIME_DIR as root (we need to write to /run here)
+		setup_user_xdg_runtime_dir(real_uid, real_gid);
+		// Ensure that the user data path exists.
+		setup_user_data(real_uid, real_gid);
+
```

Comment:
I'd prefer these be after dropping privileges. I realize you need to create /run/user/UID as root since /run/user is root:root 0755, but we should default to creating user-owned directories after dropping privileges. Please add a new function `setup_session_xdg_runtime_dir()` for creating /run/user/UID here and then move both of these back to where they were.